### DISCUSSION TITLE
Does pre sorting help

### DISCUSSION THREAD

**Discussion by djouallah** [duckdb/duckdb#4238] [Q&A] [answered]
- does pre sorting data before inserting to DuckDB tables help in performance ? if yes, which field will bring better performance ?

- Does ART improve speed on Foreign Key or it is only for PK

**Comment by hannes** [comment, marked as answer]
Yes, it can help. For example, if you order the data by a particular field and then perform a filter on the same field, duckdb can avoid reading most of the data because it has min-max indexes on every row group. But once #3913 is merged this reordering the data will even happen automatically.

**Comment by djouallah** [comment]
thanks make sense ? assuming there is no filter, does sorting PK help when doing Joins ?

**Reply by hannes** [reply]
No
